pkgs/tools/misc/pgbadger/default.nix

```
@@ -14,11 +14,18 @@ buildPerlPackage rec {
     patchShebangs ./pgbadger
   '';
 
+  # pgbadger has too many `-Idir` flags on its shebang line on Darwin,
+  # causing the build to fail when trying to generate the documentation.
+  # Rewrite the -I flags to in `use lib` form.
+  preBuild = lib.optionalString stdenv.isDarwin ''
+    shortenPerlShebang ./pgbadger
+  '';
+
   outputs = [ "out" ];
 
   PERL_MM_OPT = "INSTALL_BASE=${placeholder "out"}";
 
-  buildInputs = [ PodMarkdown JSONXS TextCSV_XS ];
+  buildInputs = [ PodMarkdown JSONXS TextCSV_XS ] ++ lib.optional stdenv.isDarwin shortenPerlShebang;
```

I think `shortenPerlShebang` should be in `nativeBuildInputs`:
```suggestion
  nativeBuildInputs = lib.optional stdenv.isDarwin shortenPerlShebang;

  buildInputs = [ PodMarkdown JSONXS TextCSV_XS ];
```

Besides that, LGTM. Thank you for your contribution!

---

Ah you're absolutely right, thanks!